### ISSUE TITLE
errno ignored for unix getaddrinfo

### ISSUE THREAD

**Issue by jwilm** [rust-lang/rust#36546]
When `getaddrinfo()` fails, the wrapper produces a custom `io::Error` using `gai_strerror()`. There's one variant which essentially says  just look at errno.

```
EAI_SYSTEM    system error returned in errno
```

But `libstd` doesn't do that which leads to very unhelpful errors.

``` rust
Io(Error { repr: Custom(Custom { kind: Other, error: StringError("failed to lookup address information: System error") }) }))
```

It would be great if the error value could be checked for `EAI_SYSTEM`, provide the `strerror()` message in the `StringError` value, and set `Kind` appropriately.

**Comment by alexcrichton** [comment]
Sounds reasonable to me! Should hopefully be more tackle-able once https://github.com/rust-lang/libc/pull/402 lands.

**Comment by apasel422** [comment]
I think this was closed by #36754.
